### DISCUSSION TITLE
Delete data at the end of the experiment

### DISCUSSION THREAD

**Discussion by ThomasChazelle** [jspsych/jsPsych#1201] [Q&A] [answered]
Hi, 
For anonymity reasons, I need to remove some identifying data I collect during an experiment before saving. The problem is that I need the data _during_ the experiment. Here is a trivial example: I collect the participant's date of birth, compute and save their age as a new property ( using addProperties() ), and then get rid of the date of birth before saving.

I guess I should use the corresponding function inside a on_finish argument in order for it to launch at the proper time.
The ignore() function doesn't seem to be working in my case, or maybe I'm not using the correct syntax.

Here is a simplified example of what I tried:
```javascript
var data_storage_notice = {
            type: 'html-keyboard-response',
            stimulus: "Only your age will be stored.",
            on_start: function(){
            jsPsych.data.ignore([
                    'rt',
                    'responses',
]) }}
```

Do you have any hint of what I should do to make this work?

Thank you all for taking part in jsPsych, you are allowing coding laymen such as me to create a whole experiment with flexibility and (overall) simplicity, and I think that is amazing!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @ThomasChazelle,

The `data.ignore()` property creates a copy of the data where the named properties have been removed. So what you can do is right before you save the data, remove all the relevant bits:

```js
// at the end of the experiment:
var data_to_save = jsPsych.data.get().ignore(['rt', 'responses']);

functionThatSavesMyData(data_to_save)
```

Another option is to modify the data at the trial level in the `on_finish` event:

```js
var trial = {
	on_finish: function(data){
		delete data.rt;
		delete data.responses;
	}
}
```

**Comment by ThomasChazelle** [comment]
Hi @jodeleeuw, 
Thanks for this clarification. Given I intend to use cognition.run I don't explicitly use a function to save the data. How should I proceed then?

**Reply by jodeleeuw** [reply]
The second solution above should work (use delete in on_finish).

**Reply by ThomasChazelle** [reply]
Thanks again!
